src/decoder/plugins/SndfileDecoderPlugin.cxx

```diff
@@ -212,7 +212,7 @@ sndfile_stream_decode(DecoderClient &client, InputStream &is)
 			break;
 
 		cmd = client.SubmitAudio(is,
-					 std::span{buffer, num_frames * frame_size},
+					 std::span{buffer, static_cast<size_t>(num_frames * frame_size)},
```

Comment:
This is the wrong way to fix this. This should never have been a signed multiplication in the first place.

Reply:
hmm? What's the proper way?

Reply:
Your code does `num_frames*frame_size` first (a signed multiply), and the result is a signed integer (which might have overflowed, which is UB), and then you cast the undefined result to unsigned (i.e. `size_t`), giving you a random value.
This is the proper way:
```
std::span{buffer, static_cast<size_t>(num_frames) * frame_size},
```
This ensures that both operands are unsigned, then does an unsigned multiply, resulting in an unsigned integer (`size_t`).

Reply:
Oh I thought you meant something else. Fixed.